### PR TITLE
Fix/264

### PR DESCRIPTION
**Pull request by Gjergj**

### CODE REVIEW

File: pkg/gw/safereference_handler_test.go [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+	}{
+		{
+			"Send correct request",
+			`{
+				"ro": {
+					"reference":  "` + refKey + `",
+					 "key": "` + refKey + `"
+				}
+			}`,
+			200,
```
Comment (mmeloni) [inline pkg/gw/safereference_handler_test.go:104]: please use meaningful constants

File: pkg/gw/safezadd_handler_test.go [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+	uknownKey := base64.StdEncoding.EncodeToString([]byte("Marias Callas"))
+	tcpPort := generateRandomTCPPort()
+	//MetricsServer must not be started as during tests because prometheus lib panis with: duplicate metrics collector registration attempted
+	op := immudb.DefaultOptions().WithPort(tcpPort).WithDir("db_" + strconv.FormatInt(int64(tcpPort), 10)).WithMetricsServer(false)
+	s := immudb.DefaultServer().WithOptions(op)
+	go s.Start()
+	time.Sleep(2 * time.Second)
+	defer func() {
+		s.Stop()
+		time.Sleep(2 * time.Second) //without the delay the db dir is deleted before all the data has been flushed to disk and results in crash.
```
Comment (mmeloni) [inline pkg/gw/safezadd_handler_test.go:30]: In order to slower down tests execution with sleeps we could add a method on immudb like store.isClosed or similar and then loop a that bool. What's your opinion?

Comment (Gjergj) [reply pkg/gw/safezadd_handler_test.go:30]: > we could add a method on immudb like store.isClosed or similar and then loop a that bool.

It's not worth it just for tests. At a later point I'd rather refactor these tests to run in parallel and two seconds for each in parallel won't add too much. I will open an issue for this so I won't forget it.
I'd say let's leave it as it is for the moment as there are not many tests.

Comment (mmeloni) [reply pkg/gw/safezadd_handler_test.go:30]: Ok perfect

**Pull request merged**